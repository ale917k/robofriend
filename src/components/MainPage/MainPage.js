import React, { useEffect } from 'react';

import CardList from '../CardList/CardList';
import SearchBox from '../SearchBox/SearchBox';
import ErrorBoundry from '../ErrorBoundary/ErrorBoundary';

import './MainPage.css';

const MainPage = ({ robots, searchField, onRequestRobots, onSearchChange, isPending }) => {
  useEffect(() => {
    onRequestRobots();
  }, [onRequestRobots]);

  const filterRobots = () => {
    return robots.filter(robot => {
      return robot.name.toLowerCase().includes(searchField.toLowerCase());
    });
  };

  return (
    <div className='tc'>
      <h1 className='f1'>RoboFriends</h1>
      <SearchBox searchChange={onSearchChange} />
      {isPending ? <h1>Loading</h1> :
        <ErrorBoundry>
          <CardList robots={filterRobots()} />
        </ErrorBoundry>
      }
    </div>
  );
}

export default MainPage;