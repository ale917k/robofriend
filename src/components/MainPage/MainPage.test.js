import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";

import React from 'react';
import MainPage from './MainPage';

describe('MainPage', () => {
    let wrapper;
    beforeEach(() => {
        const mockProps = {
            onRequestRobots: jest.fn(),
            onSearchChange: jest.fn(),
            robots: [],
            searchField: '',
            isPending: false
        };
        wrapper = shallow(<MainPage {...mockProps} />)
    });

    it('expects to match Snapshot Object', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should filter robots correctly', () => {
        const mockProps2 = {
            onRequestRobots: jest.fn(),
            onSearchChange: jest.fn(),
            robots: [{
                id: 3,
                name: 'John',
                email: 'john@gmail.com'
            }],
            searchField: 'john',
            isPending: false
        };
        const wrapper2 = shallow(<MainPage {...mockProps2} />)

        expect(wrapper).toEqual({});

        // expect(wrapper2.instance()).toEqual([{
        //     id: 3,
        //     name: 'John',
        //     email: 'john@gmail.com'
        // }]);
    });
});