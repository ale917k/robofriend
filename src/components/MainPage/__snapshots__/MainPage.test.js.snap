// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage expects to match Snapshot Object 1`] = `
<div
  className="tc"
>
  <h1
    className="f1"
  >
    RoboFriends
  </h1>
  <SearchBox
    searchChange={[MockFunction]}
  />
  <ErrorBoundary>
    <CardList
      robots={Array []}
    />
  </ErrorBoundary>
</div>
`;
